src/algebra/module/zlattice.lean

```diff
@@ -0,0 +1,224 @@
+/-
+Copyright (c) 2023 Xavier Roblot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Xavier Roblot
+-/
+import measure_theory.group.fundamental_domain
+
+/-!
+# ℤ-lattices
+
+Let `E` be a finite dimensional vector space over a `normed_linear_ordered_field` `K` with a solid
+norm and that is also a `floor_ring`, e.g. `ℚ` or `ℝ`. A (full) ℤ-lattice `L` of `E` is a discrete
+subgroup of `E` such that `L` spans `E` over `K`.
+
+The ℤ-lattice `L` can be defined in two ways:
+* For `b` a basis of `E`, then `L : submodule.span ℤ (set.range b)` is a ℤ-lattice of `E`.
+* As `L : add_subgroup E` with the additional properties:
+  `∀ r : ℝ, (L ∩ metric.closed_ball 0 r).finite`, that is `L` is discrete
+  `submodule.span ℝ (L : set E) = ⊤`, that is `L` spans `E` over `K`.
+
+## Main result
+* `zspan.is_add_fundamental_domain`: for `L : submodule.span ℤ (set.range b)` a ℤ-lattice, proves
+ that the set defined by `zspan.fundamental_domain` is indeed a fundamental domain.
+
+-/
+
+open_locale big_operators classical
```

Comment:
Nowadays `open_locale classical` is discouraged. If you need in certain proofs just use `classical` at the beginning of the proof.

Reply:
Ok. I have removed `open_locale classical`